### PR TITLE
multi: fix `lnwallet.ErrDoubleSpend`

### PR DESCRIPTION
**Pull request by yyforyongyu**
This PR fixes the matching errors used in `lnwallet.ErrDoubleSpend`, a misuse of error found in the rebroadcaster's `broadcastErrorMapper`, and updates the `btcd` version to fix another error str mismatch among different versions.

Depends on,
- https://github.com/btcsuite/btcwallet/pull/937

### CODE REVIEW

File: lnwallet/btcwallet/btcwallet.go [hunk 1]
```diff
@@ -1202,7 +1202,8 @@ func mapRpcclientError(err error) error {
 	switch {
 	// If the wallet reports a double spend, convert it to our internal
 	// ErrDoubleSpend and return.
-	case errors.Is(err, rpcclient.ErrMempoolConflict),
```
Comment (morehouse) [inline lnwallet/btcwallet/btcwallet.go:1205]: Why is `ErrMempoolConflict` handling removed?

File: go.mod [hunk 2]
```diff
@@ -208,3 +208,7 @@ replace google.golang.org/protobuf => github.com/lightninglabs/protobuf-go-hex-d
 go 1.21.4
 
 retract v0.0.2
+
+replace github.com/btcsuite/btcd => github.com/yyforyongyu/btcd v0.21.0-beta.0.20240702184211-31cf2383abd1
+
+replace github.com/btcsuite/btcwallet => github.com/yyforyongyu/btcwallet v0.16.11-0.20240702184603-cf50b410fcb1
```
Comment (morehouse) [inline go.mod:214]: I don't think we want to replace btcsuite with your fork.

**Comment by Roasbeef** [comment]
Did some `git` archeology to put together the series of changes that led to these issues, I think I understand what went wrong here, and how we can improve the state of things to avoid issues like this in the future: https://github.com/btcsuite/btcd/pull/2209#pullrequestreview-2155124809.

File: go.mod (hunk 2 above)
Comment (yyforyongyu) [reply go.mod]: 😈

File: lnwallet/btcwallet/btcwallet.go (hunk 1 above)
Comment (yyforyongyu) [reply lnwallet/btcwallet/btcwallet.go:1205]: I think we only get this error when,
1) try to fee bump a tx when it doesn't signal RBF
2) try to spend the same output in a different tx

Either way it shouldn't ever happen in ln. However I decided to not change it now to focus on fixing the error matching issue.

**Comment by yyforyongyu** [comment]
Will do a local test like [this one](https://github.com/lightningnetwork/lnd/pull/8862#issuecomment-2189586479) once the approach is ACKed, and update the release notes.

File: config_builder.go [hunk 3]
```diff
@@ -1475,27 +1478,27 @@ func parseHeaderStateAssertion(state string) (*headerfs.FilterHeader, error) {
 // the neutrino BroadcastError which allows the Rebroadcaster which currently
 // resides in the neutrino package to use all of its functionalities.
 func broadcastErrorMapper(err error) error {
-	returnErr := rpcclient.MapRPCErr(err)
+	var returnErr error
```
Comment (bhandras) [inline config_builder.go:1481]: Why not map the error here like before?

File: lnwallet/btcwallet/btcwallet.go [hunk 4]
```diff
@@ -1203,7 +1203,9 @@ func mapRpcclientError(err error) error {
 	// If the wallet reports a double spend, convert it to our internal
 	// ErrDoubleSpend and return.
 	case errors.Is(err, chain.ErrMempoolConflict),
-		errors.Is(err, chain.ErrMissingInputs):
+		errors.Is(err, chain.ErrMissingInputs),
+		errors.Is(err, chain.ErrTxAlreadyKnown),
+		errors.Is(err, chain.ErrTxAlreadyConfirmed):
```
Comment (bhandras) [inline lnwallet/btcwallet/btcwallet.go:1208]: Not sure if I'd consider `ErrTxAlreadyConfirmed` as double spend. Is there a reason to do so?

File: config_builder.go (hunk 3 above)
Comment (yyforyongyu) [reply config_builder.go:1481]: To use the method `MapRPCErr` we need to access the chain backend, which cannot be used here. Instead, we pass the already mapped error to `broadcastErrorMapper`, and access the method via `cs.MapRPCErr` above.

File: lnwallet/btcwallet/btcwallet.go (hunk 4 above)
Comment (yyforyongyu) [reply lnwallet/btcwallet/btcwallet.go:1208]: So during startup, we may rebroadcast our force close tx that can cause this error, which matches the issue voltage has. I think previously we also considered this as `ErrDoubleSpend` [here](https://github.com/btcsuite/btcwallet/pull/918/files#diff-fde7eecc7d486a2d684192946f015f42f69af8aa27e10c0a019f0f3bebb28deeL4142).

**Review by bhandras — APPROVED** [review]
LGTM 🎉

File: lnwallet/btcwallet/btcwallet.go (hunk 4 above)
Comment (ziggie1984) [reply lnwallet/btcwallet/btcwallet.go:1208]: Maybe add a comment here why we classify a already confirmed tx as doublespend ?

**Review by ziggie1984 — APPROVED** [review]
LGTM, thanks for your work 🙏

NOTE: No need to backport this because its only in LND 18.0

File: lnwallet/btcwallet/btcwallet.go [hunk 5]
```diff
@@ -1202,15 +1202,15 @@ func mapRpcclientError(err error) error {
 	switch {
 	// If the wallet reports a double spend, convert it to our internal
```
[the remark this answers is not in the document]
Comment (yyforyongyu) [reply lnwallet/btcwallet/btcwallet.go:1203]: cool updated.

File: lnwallet/btcwallet/btcwallet.go (hunk 4 above)
Comment (yyforyongyu) [reply lnwallet/btcwallet/btcwallet.go:1208]: tbh I don't think it should be considered as a double spend - we are doing it to keep the existing behavior. IMO it's better to check whether the to-be-published tx already exists in the blockchain or in the mempool before broadcasting it during startup. Or even better, understand why it's broadcast again as I believe there are some internal state updates once a tx is published, but it looks like that state was not persisted.

**Pull request merged**